packages/core/src/errors/RequestError/index.ts

```diff
@@ -29,3 +31,16 @@ export default class RequestError extends Error {
     return pick(this, 'message', 'code', 'data');
   }
 }
+
+export class RequestErrorWithUserLog extends RequestError {
+  payload: OmitAutoSetFields<UserLogDBEntry>;
+
+  constructor(
+    input: RequestErrorMetadata | LogtoErrorCode,
+    payload: OmitAutoSetFields<UserLogDBEntry>,
+    data?: unknown
+  ) {
+    super(input, data);
+    this.payload = payload;
+  }
+}
```

Comment:
thinking about an alteration of defining new error classes as well. However if we intend to use `instance of` to distinguish different error types at the `errorLogHandler` , it would be inevitable.

Reply:
reply to @gao-sun:
1. no problem.
2. `data` is used by error-handler and will be add to `ctx.body` and expose to outside, so I created `payload`. `data` is used previously for error description or error details. It's better to take a discussion in LOG-395.